### PR TITLE
docs: updated contributor guide

### LINKED ISSUE #557: doc: check and update contributor guide
- [x] Modules @musslick @younesStrittmatter 
- [x] Core @benwandrew 
- [x] Notebooks @musslick 
- [x] Development Setup @TheLemonPig (try this out while incorporating new BMS tutorial into autora repo)
- [x] Pre-Commit Hooks @benwandrew 
- [x] Remove the distinction between samplers and poolers from the "experimentalist" module doc @benwandrew 
- [x] In the index.md of the contributor guide, add the "autora-workflow" description to the autora-core description in the section "Core Contributions"

### LINKED ISSUE #625: contributor documentation must be public
Repositories of contributors that contain the documentation must be public to be incorporated into our website. We should mention this in the contributor guide.

### PR DESCRIPTION
**Pull request by musslick**
# Description

This PR updates the contributor guide to the current version of autora 4.0.0 (see detailed list of fixed below). 
- resolves #625 
- resolves #557 

# Type of change
- **docs**: Documentation only changes

# Features (Optional)
- removal of the `autora-workflow` package from the description of core packeges.
- updated examples of theorist and experimentalist implementations, primarily to be based on pandas dataframes
- added requirement for contributor repo to be public for inclusion in autora package

# Questions (Optional)
- In the modules/index.md the lists inside the Success and Hint boxes aren't rendering properly. They are just listed without a line brak. I tried changing the "-" to "*" but not sure if that fixed it? 
image

### CODE REVIEW

**Comment by hollandjg** [comment]
> * In the modules/index.md the lists inside the Success and Hint boxes aren't rendering properly. They are just listed without a line brak. I tried changing the "-" to "*" but not sure if that fixed it?

They're just missing a couple of newlines. I've pushed an extra commit which makes the block look like:
```md
!!! success
    
    In order for your package to be included in the parent package, it must meet the following criteria:
    
    * have basic documentation in ``docs/index.md``
    * have a basic python notebook exposing how to use the module in ``docs/Basic Usage.ipynb``
    * have basic tests in ``tests/``
    * be published via PyPI or Conda
    * be compatible with the current version of the parent package
    * follow standard python coding guidelines including PEP8
    * the repository in which your package is hosted must be public

The following demonstrates how ...
```

... which renders like:
Screenshot 2024-06-12 at 07 13 37

File: docs/contribute/index.md
Comment (hollandjg) [inline docs/contribute/index.md:1]: Screenshot 2024-06-12 at 07 16 41
We still have `autora-workflow` in the diagram here.

My feeling is that the dependency arrows point in the wrong direction here, but that might be a question for a different PR and a different day – I know that the way they are feels intuitive for many.

File: docs/contribute/modules/index.md [hunk 1]
```diff
@@ -27,9 +27,9 @@ After setting up your repository and linking it to your GitHub account, you can
 You may implement your code in the ``init.py`` located in the respective feature folder in ``src/autora``.
```
Comment (hollandjg) [inline docs/contribute/modules/index.md:27]: This line looks to be formatted incorrectly – it should show `__init__.py` not `init.py`
```
You may implement your code in the ``__init__.py`` located in the respective feature folder in ``src/autora``.
```

File: docs/contribute/modules/index.md [hunk 2]
```diff
@@ -27,9 +27,9 @@ After setting up your repository and linking it to your GitHub account, you can
...
 
 Please refer to the following guides on implementing
-- [theorists](theorist.md)
-- [experimentalists](experimentalist.md)
-- [experiment runners](experiment-runner.md)
+* [theorists](theorist.md)
```
Comment (hollandjg) [inline docs/contribute/modules/index.md:30]: I think you need a new line here to separate the bullet point block from the paragraph above - otherwise it won't be rendered as a bullet. It's the same problem as we had with the admonitions.

```suggestion

* [theorists](theorist.md)
```

**Review by hollandjg — CHANGES_REQUESTED** [review]
Hi Sebastian,
this looks great! The information is really clear and concise, and covers everything. 
The documentation looks really awesome.
I've a couple of very minor changes to suggest, some of which are in a separate PR. #629

File: docs/contribute/index.md
Comment (musslick) [reply docs/contribute/index.md:1]: That's a great catch! We could use the most recent figure from the paper draft. I will create an issue so we can address it in a new PR.

**Comment by musslick** [comment]
> Hi Sebastian, this looks great! The information is really clear and concise, and covers everything. The documentation looks really awesome. I've a couple of very minor changes to suggest, some of which are in a separate PR. #629

Thanks a lot John for all the fixes! I approved the request with all the fixes you implemented.

File: docs/contribute/modules/index.md [hunk 3]
```diff
@@ -27,9 +27,10 @@ After setting up your repository and linking it to your GitHub account, you can
 You may implement your code in the ``init.py`` located in the respective feature folder in ``src/autora``.
```
Comment (musslick) [inline docs/contribute/modules/index.md:27]:
```suggestion
You may implement your code in the ``__init__.py`` located in the respective feature folder in ``src/autora``.
```

File: docs/contribute/modules/index.md (hunk 1 above)
Comment (musslick) [reply docs/contribute/modules/index.md]: Thanks for the catch! I committed a fix implementing the suggested renaming from 'init.py' to '__init__.py'

**Review by hollandjg — APPROVED** [review]
LGTM!
